frame/support/src/dispatch.rs

```diff
@@ -1349,9 +1369,23 @@ macro_rules! decl_module {
 			fn on_runtime_upgrade() -> $crate::dispatch::Weight {
 				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_runtime_upgrade"));
 
-				$crate::crate_to_pallet_version!()
+				let new_storage_version = $crate::crate_to_pallet_version!();
+				new_storage_version
 					.put_into_storage::<<$trait_instance as $system::Config>::PalletInfo, Self>();
 
+				let pallet_name = <<
+					$trait_instance
+					as
+					$system::Config
+				>::PalletInfo as $crate::traits::PalletInfo>::name::<Self>().expect("pallet will have name in the runtime.");
```

Comment:
Same as above and what is with the new proc macros?

Reply:
ahh the new proc macro should go away now, that is old.